Src/Particle/AMReX_ArrayOfStructs.H

```diff
@@ -62,16 +62,10 @@ public:
         resize(nrp + num_neighbors);
     }
 
-    [[nodiscard]] int getNumNeighbors () { return m_num_neighbor_particles; }
+    [[nodiscard]] int getNumNeighbors () const { return m_num_neighbor_particles; }
 
     [[nodiscard]] bool empty () const { return m_data.empty(); }
 
-    [[nodiscard]] const RealType* data () const { return &(m_data[0].m_pos[0]); }
```

Comment:
I agree that these are weird, but unfortunately they are needed for backwards compatibility with codes that still pass particle structs into Fortran subroutines.

Reply:
Should we potentially make the `void*` pointers to reflect what they do?

(Can be a separate PR.)

Reply:
Ah I see, changed to `ParticleType*` is also good.